src/main/java/com/lizin5ths/indypets/mixin/TameableEntityMixin.java

```diff
@@ -59,11 +59,11 @@ private static NbtList toNbtList(int... values) {
 
 	@Inject(method = "readCustomDataFromNbt", at = @At("TAIL"))
 	private void indypets$readFollowDataFromNbt(NbtCompound nbt, CallbackInfo callbackInfo) {
-		indypets$isIndependent = !nbt.getBoolean("AllowedToFollow");
+		indypets$isIndependent = !nbt.getBoolean("AllowedToFollow").orElse(false);
 
-		if (nbt.contains("IndyPets$HomePos", NbtElement.LIST_TYPE)) {
-			NbtList nbtList = nbt.getList("IndyPets$HomePos", NbtElement.INT_TYPE);
-			indypets$homePos = new BlockPos(nbtList.getInt(0), nbtList.getInt(1), nbtList.getInt(2));
+		if (nbt.contains("IndyPets$HomePos")) {
+			NbtList nbtList = nbt.getList("IndyPets$HomePos").get();
+			indypets$homePos = new BlockPos(nbtList.getInt(0).get(), nbtList.getInt(1).get(), nbtList.getInt(2).get());
```

Comment:
I'd probably rewrite this somehow, not entirely sure how.
Maybe using `orElseThrow()` would make the semantics more clear (and silence the IDE).
Maybe something like this
```
Optional<NbtList> nbtList = nbt.getList("IndyPets$HomePos");
if (nbtList.isPresent()) {
	NbtList homePos = nbtList.get();
	indypets$homePos = new BlockPos(homePos.getInt(0).orElseThrow(), homePos.getInt(1).orElseThrow(), homePos.getInt(2).orElseThrow());
}
```

Yeah, no... I'll think about it. (Don't worry about it.)